File: Data Structures & Algorithms/duplicate-integer/submission-5.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        mp = {}
        for i in nums:
            mp[i] = mp.get(i,0)+1
        
        for i in mp:
            if mp.get(i)>1:
                return True       
        return False

File: Data Structures & Algorithms/contains-duplicate-ii/submission-5.java
class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        HashMap<Integer,Integer> map = new HashMap<>();
        int n = nums.length;
        for(int i=0;i<n;i++){
            if(map.containsKey(nums[i])){
                int check = Math.abs(map.get(nums[i])-i);
                if(check<=k){
                    return true;
                }
            }


                  map.put(nums[i],i);
  
  
          
        }
        return false;

    }
}